.atomist/editors/ProjectToGeneratorOperations.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2017 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Project, File } from '@atomist/rug/model/Core'
+import { PathExpression, PathExpressionEngine } from '@atomist/rug/tree/PathExpression'
+
+export function addAssertionsForAllFilesInProject(project: Project, generatorName: string): void {
+    let originalTestAssertionsString = "then\n  fileExists \"README.md\""
+
+    let testAssertionsString = project.files().reduce(function(acc, file) {
+      if (notACommonPerpheralArtifact(file))  {
+          if(acc === "then"){
+            acc +=  "\n  "
+          }else{
+            acc +=  "\n    and "
+          }
+          acc += "fileExists \"" + file.path() + "\"";
+      }
+      return acc;
+    },"then")
+
+    let eng: PathExpressionEngine = project.context().pathExpressionEngine();
+    let testPathExpression = new PathExpression<Project, File>("/*[@name='.atomist']/tests/*[@name='" + generatorName + ".rt']");
+    let testFile: File = eng.scalar(project, testPathExpression);
+    testFile.replace(originalTestAssertionsString, testAssertionsString)
+}
+
+const pathToAtomist = ".atomist";
+const pathToNodeModules = "node_modules";
+const pathToTarget = "target";
+const pathToIdeaIDEArtifacts = ".idea";
```

Comment:
Why define these constants here? The only one that is used is `pathToAtomist` and it is only used in `notACommonPeripheralArtifact`.